tempodb/encoding/vparquet4/block_traceql.go

```diff
@@ -230,27 +239,37 @@ func (s *span) AttributeFor(a traceql.Attribute) (traceql.Static, bool) {
 
 	// name search in span, resource, link, and event to give precedence to span
 	// we don't need to do a name search at the trace level b/c it is intrinsics only
-	if attr := findName(a.Name, s.spanAttrs); attr != nil {
-		return *attr, true
+	if len(s.spanAttrs) > 0 {
```

Comment:
Why don't we move this logic to the findName method instead? We are already counting the length. We can add a first case:

```go
if len(attrs) == 0 {
return nil
}
```

We are adding a new entry in the call stack but the code is nicer

Reply:
Right, this way it's skipping the method call entirely, so it ends up faster.  I think it's worth it, even though the code is less nice.